docs/source/templates/class_api.rst

```diff
@@ -0,0 +1,197 @@
+
+..
+  The (pulic header) file the user will include in their code
+
+Header File: ``{{ core_header_file }}``
+
+..
+  High-level, human-language summary of what the thing does, and if possible, brief statement about why it exists (2 - 3 sentences, max);
+
+Description
+-----------
+
+..
+  The API of the entity.
+
+Interface
+---------
+
+..
+  The declaration or signature of the entity.
+
+.. cppkokkos:class:: template <class DataType, class... Traits> CoolerView
+
+  ..
+    Template parameters (if applicable)
+    Omit template parameters that are just used for specialization/is deduced/ and/or should not be exposed to the user.
+
+  .. rubric:: Template Parameters
+
+  :tparam Foo: Description of the Foo template parameter
+
+  ..
+    Parameters (if applicable)
+
+  .. rubric:: Parameters
+
+  :param bar: Description of the bar parameter
+
+  .. rubric:: Public Types
+
+  .. cppkokkos:type:: data_type
+
+    Some interesting description of the type and how to use it.
+
+  .. rubric:: Static Public Member Variables
+
+  .. cppkokkos:member:: int some_var = 5;
+
+    Description of some_var
+
+    ..
+      If you have related info
+
+    .. seealso::
+
+      ..
+        We can cross-reference entities
+
+      The :cppkokkos:func:`frobrnicator` free function.
+
+  .. rubric:: Constructor
+
+  .. cppkokkos:function:: CoolerView(CoolerView&& rhs)
+
+    Whether it's a move/copy/default constructor. Describe what it does.
+
+  ..
+    Only include the destructor if it does something interesting as part of the API, such as RAII classes that release a resource on their destructor. Classes that merely
+    clean up or destroy their members don't need this member documented.
+
+  .. rubric:: Destructor
+
+  .. cppkokkos:function:: ~CoolerView()
+
+    Document what special effect the destructor has.
+
+  .. rubric:: Public Member Functions
+
+  .. cppkokkos:function:: template<class U> foo(U x)
+
+    Brief description of the function.
+
+    :tparam U: Description of U
+
+    :param: description of x
+
+    ..
+      Describe any API changes between versions.
+
+    .. versionchanged:: 3.7.1
+
+      Only takes one parameter for foo-style operations instead of two.
+
+  ..
+    Deprecated types need special handling, both for the badge displayed beforehand and the deprecated version directive.
+    Note, there is not currently any functionality for other deprecated entities.
+
+  .. cppkokkos:deprecated-type:: 4.0.1 foobar
+
+    Represents the foobar capability.
+
+    .. deprecated:: 4.0.1
+
+      Use :cppkokkos:type:`foobat` instead.
+
+  .. cppkokkos:type:: foobat
+
+    A better version of foobar.
+
+    .. versionadded:: 4.0.1
+
+
+Non-Member Functions
+----------------------
+
+..
+  These should only be listed here if they are closely related. E.g. friend operators. However,
+  something like view_alloc shouldn't be here for view
+
+.. cppkokkos:function:: template<class ViewSrc> bool operator==(CoolerView, ViewSrc);
+
```

Comment:
Is this an actual comparator?  i.e., something testing that CoolerView is the same as ViewSrc, or is this just your example of an "allied" or "friend" operator?

Reply:
Just an example